core/cmd/remote_client.go

```diff
@@ -301,48 +301,55 @@ func (cli *Client) Profile(c *clipkg.Context) error {
 		"trace",        // A trace of execution of the current program.
 	}
 	wgPprof.Add(len(vitals))
-	errs := make(chan error)
+	cli.Logger.Infof("Collecting profiles: %v", vitals)
+	cli.Logger.Infof("writing debug info to %s", genDir)
+
+	errs := make(chan error, len(vitals))
 	for _, vt := range vitals {
 		go func(vt string) {
 			defer wgPprof.Done()
 			uri := fmt.Sprintf("/v2/debug/pprof/%s?seconds=%d", vt, seconds)
-			cli.Logger.Infof("Collecting %s ", uri)
 			resp, err := cli.HTTP.Get(uri)
 			if err != nil {
-				cli.Logger.Errorf("error collecting vt %s: %s", vt, err.Error())
-				errs <- err
+				errs <- errors.Wrapf(err, "error collecting %s", vt)
+				return
+			}
+			if resp.StatusCode == http.StatusUnauthorized {
+				errs <- errors.Wrapf(errUnauthorized, "error collecting %s", vt)
 				return
 			}
 			defer resp.Body.Close()
-
 			// write to file
 			f, err := os.Create(filepath.Join(genDir, vt))
 			if err != nil {
-				cli.Logger.Errorf("error creating file for %s: %s", vt, err.Error())
-				errs <- err
+				errs <- errors.Wrapf(err, "error creating file for %s", vt)
 				return
 			}
 			wc := utils.NewDeferableWriteCloser(f)
 			defer wc.Close()
 
 			_, err = io.Copy(wc, resp.Body)
 			if err != nil {
-				cli.Logger.Errorf("error writing to file for %s: %s", vt, err.Error())
-				errs <- err
+				errs <- errors.Wrapf(err, "error writing to file for %s", vt)
 				return
 			}
-			cli.Logger.Infof("Collected %s", vt)
 			err = wc.Close()
 			if err != nil {
-				cli.Logger.Errorf("error closing file for %s: %s", vt, err.Error())
-				errs <- err
+				errs <- errors.Wrapf(err, "error closing file for %s", vt)
 				return
 			}
 		}(vt)
 	}
 	wgPprof.Wait()
+	close(errs)
+	// Atmost one err is emitted per vital.
+	cli.Logger.Infof("collected %d/%d profiles", len(vitals)-len(errs), len(vitals))
 	if len(errs) > 0 {
-		return cli.errorOut(fmt.Errorf("%d profile collections failed", len(errs)))
+		var merr error
+		for err := range errs {
+			merr = multierr.Append(merr, err)
```

Comment:
`errors.Join`? :shrug: 
```suggestion
			merr = errors.Join(merr, err)
```